release_connection after cleaning

nerdrew opened the pull request:
For Rails >= 7.2, this library gets a connection via `#lease_connection`. But it never releases the connection. Release the connection back to the pool after `#clean`-ing.

Rails >= 7.2 does connection health checks / reconnect on connection checkout. If the connection is never checked back into the pool, it will never get "repaired", which makes testing database connection issues tricky.

etagwerker reviewed (approved):
@nerdrew Looks good, could you please add an entry to the CHANGELOG.md file?

nerdrew committed:
release_connection after cleaning

For Rails >= 7.2, this library gets a connection via
`#lease_connection`. But it never releases the connection. Release the
connection back to the pool after `#clean`-ing.

Rails >= 7.2 does connection health checks / reconnect on connection
checkout. If the connection is never checked back into the pool, it will
never get "repaired", which makes testing database connection issues
tricky.

nerdrew commented:
Added a changelog entry.
